compiler/wasm/src/compile_new.rs

```diff
@@ -0,0 +1,258 @@
+use crate::compile::{
+    file_manager_with_source_map, preprocess_contract, preprocess_program, JsCompileResult,
+    PathToFileSourceMap,
+};
+use crate::errors::{CompileError, JsCompileError};
+use noirc_driver::{
+    add_dep, compile_contract, compile_main, prepare_crate, prepare_dependency, CompileOptions,
+};
+use noirc_frontend::{
+    graph::{CrateGraph, CrateId, CrateName},
+    hir::Context,
+};
+use std::path::Path;
+use wasm_bindgen::prelude::wasm_bindgen;
+
+#[wasm_bindgen]
+pub struct ContextWrapper {
+    context: Context,
```

Comment:
Having this means we also have a CrateGraph instance which means we do not need the JsDependencyGraph struct anymore.

The downside is that someone needs to manage the CrateId's that get returned -- if we want to fix this, I think we should modify the code to share more in common with nargo's methodology for managing this see `prepare_dependencies` -- The issue with that code is that it is resolving remote/local dependencies which I think needs to be modified to allow us to share more code there.